notify/grafana_alertmanager.go

```diff
@@ -96,10 +95,9 @@ type GrafanaAlertmanager struct {
 	// buildReceiverIntegrationsFunc builds the integrations for a receiver based on its APIReceiver configuration and the current parsed template.
 	buildReceiverIntegrationsFunc func(next *APIReceiver, tmpl *templates.Template) ([]*Integration, error)
 	externalURL                   string
-	workingDirectory              string
 
-	// templates contains the filenames (not full paths) of the persisted templates that were used to construct the current parsed template.
-	templates []string
+	// templates contains the template name -> template contents for each user-defined template.
+	templates []templates.TemplateDefinition
```

Comment:
nit: Is this better off as a map? I don't see a reason for introducing another type.

Reply:
I originally implemented as a map, but looking at GrafanaAlertmanager as a library I felt that being able to choose and guarantee ordering when passing in templates was best. It also happens to mimic the structure of the existing [TemplateDesc](https://github.com/grafana/mimir/blob/bc34d2d8ee4b7b2dfc57cb0a828188fba59ebe74/pkg/alertmanager/alertspb/alerts.pb.go#L88) proto in Mimir which, while not necessary, is nice.